context.go

```diff
@@ -88,11 +96,14 @@ func (c SpanContext) IsDebug() bool {
 
 // IsValid indicates whether this context actually represents a valid trace.
 func (c SpanContext) IsValid() bool {
-	return c.traceID != 0 && c.spanID != 0
+	return c.traceID.IsValid() && c.spanID != 0
```

Comment:
what was the decision to not to have and a separate function for SpanID.isValid()? user's don't need to know that 0 means invalid span id.

Also, is it possible that when we generate a trace or span id that we can get 0? or do we loop if it is 0?